src/client/components/app/Transactions/ClaimRevenueTx.tsx

```diff
@@ -0,0 +1,348 @@
+          exit={onTransactionCompletedDismissed}
+        />
+      </StyledTransaction>
+    );
+  }
+
+  if (transactionCompleted === 2) {
+    return (
+      <StyledTransaction onClose={onClose} header={'transaction'}>
+        <TxStatus
+          success={transactionCompleted}
+          transactionCompletedLabel={'could not add credit'}
+          exit={onTransactionCompletedDismissed}
+        />
+      </StyledTransaction>
+    );
+  }
+
+  if (!selectedSellToken && !selectedTokenAddress) return null;
+
+  // TODO: Autopopulate claimFunc in modal so this function can be deprecated.
+  const onClaimFuncSelection = (newFunc: { id: string; label: string; value: string }) => {
+    const funcInputs = generateClaimFuncInputs(newFunc.label, contractABI[selectedRevenueContract]!);
+    console.log('Func Inputs: ', funcInputs);
+    setFuncInputs(funcInputs);
+    const hashedSigFunc = generateSig(newFunc.label, contractABI[selectedRevenueContract]!);
+    setClaimFunc(hashedSigFunc);
+    setClaimFuncType(newFunc);
+
+    // generateClaimFuncInputs - setSlaimFuncInputs into component state
+    // add input fields to modal
+  };
+
+  const isVerifiedContract =
+    isValidAddress(selectedRevenueContract) &&
+    contractABI &&
+    selectedContractFunctions[selectedRevenueContract] !== undefined &&
+    Array.from(Object.values(selectedContractFunctions[selectedRevenueContract]))!.length !== 0;
+
+  const funcOptions = !selectedContractFunctions[selectedRevenueContract]
+    ? []
+    : Array.from(Object.values(selectedContractFunctions[selectedRevenueContract])).map((func, i) => ({
+        id: i.toString(),
+        label: func,
+        value: '',
+      }));
+  console.log('Func Options: ', funcOptions);
+
+  const handleInputChange = (name: string, type: string, value: any) => {
+    console.log('handle input change', name, value);
+    let newUserFuncInputs = { ...userFuncInputs };
+    newUserFuncInputs[name] = value;
+    setUserFuncInputs(newUserFuncInputs);
+    console.log('handle input change: ', userFuncInputs);
+  };
+
+  const generateInputFields = () => {
+    if (funcInputs.length === 0) {
+      return <></>;
+    }
+    const inputFieldsHTML = [];
+    for (let i = 0; i < funcInputs.length; i++) {
+      const input = funcInputs[i];
+      if (input.type.includes('int')) {
+        inputFieldsHTML.push(
+          <TxNumberInput
+            headerText={input.name}
+            amount={userFuncInputs[input.name] ?? ''}
+            placeholder={'0'}
+            onInputChange={(e: any) => handleInputChange(input.name, input.type, e)}
+          />
+        );
+      }
+      if (input.type === 'address') {
+        inputFieldsHTML.push(
+          <TxAddressInput
+            headerText={input.name}
+            address={userFuncInputs[input.name] ?? ''}
+            onAddressChange={(e: any) => handleInputChange(input.name, input.type, e)}
+          />
+        );
+      }
+      if (input.type === 'bytes') {
```

Comment:
does this work for all byte types? maybe do `.includes()` too

Reply:
fixed

Reply:
fixed?